ceph/ceph_admin/helper.py

```diff
@@ -257,11 +258,35 @@ def generate_osd_spec(self, spec):
                     all: boolean      # true or false
                 encrypted: boolean    # true or false
 
+
+            spec:
+              - service_type: osd
+                unmanaged: boolean    # true or false
+                placement:
+                  host_pattern: "*"   # either hosts or host_pattern
+                  nodes:
+                    - node2
+                    - node3
+                spec:
+                    data_devices:
+                        paths: true       # true or false
+                    encrypted: boolean    # true or false
+                extra_container_args:
+                    - "--cpus=2"
         """
         template = self._get_template("osd")
         node_names = spec["placement"].pop("nodes", None)
+
         if node_names:
             spec["placement"]["hosts"] = self.get_hostnames(node_names)
+        for item, values in spec["spec"].items():
+            if item in ["data_devices", "db_devices", "wal_devices"]:
+                if values.get("paths"):
```

Comment:
likewise, if `paths == "all"` then we replace the disk_devs to all known volumes in the node else set the disk_devs to the values provided in paths.